### DISCUSSION TITLE
ccache with MSBuild (MSVC compiler) with precompile header

### DISCUSSION THREAD

**Discussion by mohsin-TW** [ccache/ccache#1253] [Questions & answers]
Does ccache with MSVC compiler work with precompiled headers? I tried but it did not work for me. Is there any flag/configuration that needs to be set for ccache to work with MSVC with precompiled headers?

**Comment by md-pix4d** [comment]
I am currently fighting with this to try to make it work using the latest (4.8) version of ccache. Here is what I can say for now:

- caching the precompiled header generation itself is currently not supported by ccache (the `/Yc` option is treated as `TOO_HARD`), so the PCH file will always be regenerated. These are listed as `not cacheable` in the statistics
- my current experiments with "just" standard library headers in my PCH file shows that regenerating them results in cache misses because of different hashes, simply because the actual `cmake_pch.cxx.pch` file generated is different. As you can see I am using CMake to configure precompiled headers. My tests were conducted with these tools:

```
Microsoft Visual Studio Professional 2019 - Version 16.11.19
cmake version 3.24.2
```

I am currently trying other options to see if I can workaround these limitations and at least get cache hits when the contents of the actual PCH file does not change, but no luck so far. Any advice would be greatly appreciated!
